services/core-web/src/components/noticeOfWork/applications/permitGeneration/NOWPermitGeneration.js

```diff
@@ -1,29 +1,126 @@
 import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
 import { formatDate } from "@common/utils/helpers";
+import { getFormValues, reset } from "redux-form";
 import { getNoticeOfWorkApplicationTypeOptions } from "@common/selectors/staticContentSelectors";
+import {
+  createPermit,
+  fetchPermits,
+  createPermitAmendment,
+  updatePermitAmendment,
+  fetchDraftPermitByNOW,
+} from "@common/actionCreators/permitActionCreator";
+import { getPermits, getDraftPermitForNOW } from "@common/selectors/permitSelectors";
+import * as FORM from "@/constants/forms";
 import CustomPropTypes from "@/customPropTypes";
-import GeneratePermitForm from "@/components/Forms/GeneratePermitForm";
+import GeneratePermitForm from "@/components/Forms/permits/GeneratePermitForm";
+import PreDraftPermitForm from "@/components/Forms/permits/PreDraftPermitForm";
+import NullScreen from "@/components/common/NullScreen";
+import * as routes from "@/constants/routes";
+import NOWSideMenu from "@/components/noticeOfWork/applications/NOWSideMenu";
+import LoadingWrapper from "@/components/common/wrappers/LoadingWrapper";
 
 /**
  * @class NOWPermitGeneration - contains the form and information to generate a permit document form a Notice of Work
  */
 
 const propTypes = {
-  returnToPrevStep: PropTypes.func.isRequired,
   noticeOfWork: CustomPropTypes.importedNOWApplication.isRequired,
   appOptions: PropTypes.arrayOf(CustomPropTypes.options).isRequired,
   handleGenerateDocumentFormSubmit: PropTypes.func.isRequired,
   documentType: PropTypes.objectOf(PropTypes.any).isRequired,
+  toggleEditMode: PropTypes.func.isRequired,
+  isViewMode: PropTypes.bool.isRequired,
+  fixedTop: PropTypes.bool.isRequired,
+  reset: PropTypes.func.isRequired,
+  createPermit: PropTypes.func.isRequired,
+  fetchPermits: PropTypes.func.isRequired,
+  createPermitAmendment: PropTypes.func.isRequired,
+  updatePermitAmendment: PropTypes.func.isRequired,
+  fetchDraftPermitByNOW: PropTypes.func.isRequired,
+  formValues: CustomPropTypes.permitGenObj.isRequired,
+  preDraftFormValues: CustomPropTypes.preDraftForm.isRequired,
+  permits: PropTypes.arrayOf(CustomPropTypes.permit).isRequired,
+  draftPermit: CustomPropTypes.permit.isRequired,
   isAmendment: PropTypes.bool.isRequired,
 };
 
 const defaultProps = {};
 
+const originalPermit = "OGP";
+const draft = "DFT";
+
 export class NOWPermitGeneration extends Component {
-  createPermitGenObject = (noticeOfWork) => {
+  state = {
+    isPreDraft: false,
+    isDraft: false,
+    permittee: {},
+    draftAmendment: {},
+    permitGenObj: {},
+    isLoaded: false,
+  };
+
+  componentDidMount() {
+    const permittee = this.props.noticeOfWork.contacts.filter(
+      (contact) => contact.mine_party_appt_type_code_description === "Permittee"
+    )[0];
+    this.setState({ permittee });
+    this.props.fetchPermits(this.props.noticeOfWork.mine_guid);
+    this.handleDraftPermit();
+  }
+
+  handleDraftPermit = () => {
```

Comment:
If we continue working on this file we should probably break some of these out into other files as stateless components, as it's getting pretty big.